pepper-apis/src/main/java/org/broadinstitute/ddp/cache/LanguageStore.java

```diff
@@ -24,7 +24,7 @@ public static LanguageDto getDefault() {
         return languages.get(DEFAULT_LANG_CODE);
     }
 
-    public static LanguageDto getOrCompute(Handle handle, String isoCode) {
+    public static synchronized LanguageDto getOrCompute(Handle handle, String isoCode) {
```

Comment:
We're not exposing the backing HashMap to callers so there's no collection to iterate over? We do have the `get()` methods that I purposely did not `synchronized` -- I was thinking it will be fine if caller is just doing a "read" while other threads is mutating underlying map so we save some synchronization costs, but if this is not recommended let me know and I'll tack on `synchronized` on those as well!

Reply:
The list of languages is a pretty small list, so what we can do alternatively is load up the whole list once and make the backing HashMap immutable. That should also solve concurrency issues.

Reply:
That'd be great! I'm onboard with that fix

Reply:
Do we think we'd ever end up with enough languages that this approach would be worth it?  If not, it seems like loading the languages once might also simplify the code, which would be nice.

Reply:
I think there's like 100 or so of these? Which is small potatoes...

Reply:
30 files changed!  I can see why this wasn't your original solution

Reply:
Actually, I wasn't farsighted enough to predict the 30 files -- rather, I was basing my original design choice on how we have done other similar "cache stores", which is to use synchronization blocks. But thinking more about this particular one, it appears that its nature/lifetime is a bit different. The 30 files is just a manifestation of how widespread we're querying the languages!